### PR TITLE
Added events example

### PR DESCRIPTION
**Pull request by teubert**

### CODE REVIEW

File: examples/events.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+Example further illustrating the concept of 'events' which generalizes the idea of EOL. 
+
+'Events' is the term used to describe something to be predicted. 
+Generally in the PHM community these are referred to as End of Life (EOL). 
+However, they can be much more.
+
+In the implementation from the prog_models package, events can be anything that needs to be predicted. 
+Events can represent end of life (EOL), end of mission (EOM), warning thresholds, or any event of interest (EOI). 
+
+This example shows how a events can be used in your applications.
```
Comment (mollocon44) [inline examples/events.py:14]: typo - "a events"

File: examples/events.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+This example shows how a events can be used in your applications. 
+"""
+from prog_models.models import BatteryElectroChemEOD
+
+def run_example():
+    # Example: Warning thresholds
+    # In this example we will use the battery model
+    # We of course are interested in end of discharge, but for this example we
+    # have a requirement that says the battery must not fall below 5% SOC
```
Comment (mollocon44) [inline examples/events.py:22]: I don't think you've explained SOC==State of Charge yet

File: examples/events.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    # Additionally, we have two warning thresholds (yellow and red)
+
+    YELLOW_THRESH = 0.15
```
Comment (mollocon44) [inline examples/events.py:25]: This would be a great spot to remind/explain that the battery SOC goes from 0 to 1 - I think I was confused by that at first

File: examples/events.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    # Step 1: Extend the battery model to define the additional events
+    class MyBatt(BatteryElectroChemEOD):
+        events = BatteryElectroChemEOD.events + ['EOD_warn_yellow', 'EOD_warn_red']
+
+        def event_state(self, state):
+            # Get event state from parent
+            event_state = super().event_state(state)
+
+            # Add yellow and red states by scaling EOD state
```
Comment (mollocon44) [inline examples/events.py:36]: This is a little hard to follow - is it scaling to make the EOD effectively 0% SOC?

File: examples/events.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            # Add yell and red states from event_state
+            event_state = self.event_state(x)
+            t_met['EOD_warn_yellow'] = event_state['EOD_warn_yellow'] <= 0
+            t_met['EOD_warn_red'] = event_state['EOD_warn_red'] <= 0
+
+            return t_met
+
+    # Step 2: Use it
+    # 2a: Setup model
+    def future_loading(t, x=None):
```
Comment (mollocon44) [inline examples/events.py:56]: Might want to remind user that this is future current loading

File: examples/events.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            i = 1
+        elif (t < 1800):
+            i = 4
+        elif (t < 3000):
+            i = 2     
+        else:
+            i = 3
+        return {'i': i}
+    
+    first_output = {'t': 18.95, 'v': 4.183}
```
Comment (mollocon44) [inline examples/events.py:70]: Is first_output the initial state of the battery?

**Review by mollocon44 — APPROVED** [review]
A really nice example overall - most of my comments are on leaving extra comments, with just two spots where I was a little confused  about what the code was doing (37/38 and 70)

File: examples/events.py (hunk 4 above)
Comment (mollocon44) [reply examples/events.py:36]: Ohhhh it's scaling between voltage levels and percentages, isn't it. This could probably use an extra comment, I got a bit lost here.

File: examples/events.py (hunk 6 above)
Comment (teubert) [reply examples/events.py:70]: It's the first output (i.e., measurement), used to estimate first state. It's not strictly needed here and doesn't add anything to the example, so I'm going to remove it.

File: examples/events.py [hunk 7]
```diff
@@ -67,11 +76,10 @@ def future_loading(t, x=None):
             i = 3
         return {'i': i}
     
-    first_output = {'t': 18.95, 'v': 4.183}
     m = MyBatt()
 
     # 2b: Simulate to threshold
-    (times, inputs, states, outputs, event_states) = m.simulate_to_threshold(future_loading, first_output, threshold_keys=['EOD'], print = True)
+    (_, _, _, _, event_states) = m.simulate_to_threshold(future_loading, threshold_keys=['EOD'], print = True)
```
Comment (mollocon44) [inline examples/events.py:82]: For clarity, it might be nice to leave (times, inputs, states, outputs, event_states) as a reminder of what the output is, unless the code needs it to be shorter

Comment (teubert) [reply examples/events.py:82]:
```suggestion
    (times, inputs, states, outputs, event_states) = m.simulate_to_threshold(future_loading, threshold_keys=['EOD'], print = True)
```
